src/PlotlyEditor.js

```diff
@@ -35,23 +44,95 @@ class PlotlyEditor extends Component {
       graphDiv: gd,
       layout: gd.layout,
       locale: this.props.locale,
-      onUpdate: this.updateProp.bind(this),
+      onUpdate: this.handleUpdate.bind(this),
       plotSchema: this.plotSchema,
       plotly: this.props.plotly,
     };
   }
 
-  updateProp(event) {
+  handleUpdate({type, payload}) {
```

Comment:
We could go one step further here and split this handler into `handleUpdateTraces`, `handleDeleteAnnotation` etc... there's essentially no shared code here and no real need to have a standard `{type, payload}` pattern IMO.

Reply:
Is there a case where users will want to have their own update function? What if they want the updates to happen inside some reducer file, or apply updates and also do other things... If there are multiple functions it may be more difficult to replace?